packages/core/src/Serialization/TransactionBody/TransactionBody.ts

```diff
@@ -924,11 +924,51 @@ export class TransactionBody {
     return this.#donation;
   }
 
+  /**
+   * Computes the hash of the transaction body.
+   *
+   * @returns The hash of the transaction body.
+   */
   hash() {
     const hash = Crypto.blake2b(Crypto.blake2b.BYTES).update(hexToBytes(this.toCbor())).digest();
     return TransactionId.fromHexBlob(HexBlob.fromBytes(hash));
   }
 
+  /**
+   * Checks if the transaction body has tagged sets.
+   *
+   * @returns true if the transaction body has tagged sets, false otherwise.
+   */
+  hasTaggedSets() {
+    if (this.#inputs.size() > 0) {
+      return TransactionBody.startsWithTag(this.#inputs.toCbor());
+    }
+
+    if (this.#requiredSigners?.size()) {
+      return TransactionBody.startsWithTag(this.#requiredSigners.toCbor());
+    }
+
+    if (this.#certs?.size()) {
+      return TransactionBody.startsWithTag(this.#certs.toCbor());
+    }
+
+    if (this.#proposalProcedures?.size()) {
+      return TransactionBody.startsWithTag(this.#proposalProcedures.toCbor());
+    }
```

Comment:
This can be simplified as transactions must have at least 1 input
```suggestion
```

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1460/commits/948366e517184ab23c9443a32a9173beecb4776d